### PR TITLE
subadmin methods should not return any null user or group

### LINKED ISSUE #20070: PHP fatal in subadmin
@rullzer 

````
Mon Oct 26 22:06:54 2015] PHP Fatal error:  Call to a member function getUID() on null in /home/deepdiver/Development/ownCloud/master/lib/private/subadmin.php on line 190
[Mon Oct 26 22:06:54 2015] PHP Stack trace:
[Mon Oct 26 22:06:54 2015] PHP   1. {main}() /home/deepdiver/Development/ownCloud/master/index.php:0
[Mon Oct 26 22:06:54 2015] PHP   2. OC::handleRequest() /home/deepdiver/Development/ownCloud/master/index.php:39
[Mon Oct 26 22:06:54 2015] PHP   3. OC\Route\Router->match() /home/deepdiver/Development/ownCloud/master/lib/base.php:852
[Mon Oct 26 22:06:54 2015] PHP   4. call_user_func:{/home/deepdiver/Development/ownCloud/master/lib/private/route/router.php:282}() /home/deepdiver/Development/ownCloud/master/lib/private/route/router.php:282
[Mon Oct 26 22:06:54 2015] PHP   5. OC\AppFramework\routing\RouteActionHandler->__invoke() /home/deepdiver/Development/ownCloud/master/lib/private/route/router.php:282
[Mon Oct 26 22:06:54 2015] PHP   6. OC\AppFramework\App::main() /home/deepdiver/Development/ownCloud/master/lib/private/appframework/routing/routeactionhandler.php:45
[Mon Oct 26 22:06:54 2015] PHP   7. Pimple\Container->offsetGet() /home/deepdiver/Development/ownCloud/master/lib/private/appframework/routing/routeactionhandler.php:100
[Mon Oct 26 22:06:54 2015] PHP   8. OC\AppFramework\DependencyInjection\DIContainer->OC\AppFramework\DependencyInjection\{closure}() /home/deepdiver/Development/ownCloud/master/3rdparty/pimple/pimple/src/Pimple/Container.php:113
[Mon Oct 26 22:06:54 2015] PHP   9. Pimple\Container->offsetGet() /home/deepdiver/Development/ownCloud/master/3rdparty/pimple/pimple/src/Pimple/Container.php:265
[Mon Oct 26 22:06:54 2015] PHP  10. OC\AppFramework\DependencyInjection\DIContainer->OC\AppFramework\DependencyInjection\{closure}() /home/deepdiver/Development/ownCloud/master/3rdparty/pimple/pimple/src/Pimple/Container.php:113
[Mon Oct 26 22:06:54 2015] PHP  11. Pimple\Container->offsetGet() /home/deepdiver/Development/ownCloud/master/3rdparty/pimple/pimple/src/Pimple/Container.php:312
[Mon Oct 26 22:06:54 2015] PHP  12. OC\Settings\Application->OC\Settings\{closure}() /home/deepdiver/Development/ownCloud/master/3rdparty/pimple/pimple/src/Pimple/Container.php:113
[Mon Oct 26 22:06:54 2015] PHP  13. OC\AppFramework\Utility\SimpleContainer->query() /home/deepdiver/Development/ownCloud/master/settings/application.php:169
[Mon Oct 26 22:06:54 2015] PHP  14. Pimple\Container->offsetGet() /home/deepdiver/Development/ownCloud/master/lib/private/appframework/utility/simplecontainer.php:104
[Mon Oct 26 22:06:54 2015] PHP  15. OC\Settings\Application->OC\Settings\{closure}() /home/deepdiver/Development/ownCloud/master/3rdparty/pimple/pimple/src/Pimple/Container.php:113
[Mon Oct 26 22:06:54 2015] PHP  16. OC_SubAdmin::isSubAdmin() /home/deepdiver/Development/ownCloud/master/settings/application.php:202
[Mon Oct 26 22:06:54 2015] PHP  17. OC\SubAdmin->isSubAdmin() /home/deepdiver/Development/ownCloud/master/lib/private/legacy/subadmin.php:143
[Mon Oct 26 22:06:54 2015] ::1:46908 [500]: /index.php/settings/apps - Call to a member function getUID() on null in /home/deepdiver/Development/ownCloud/master/lib/private/subadmin.php on line 190

### PR DESCRIPTION
**Pull request by blizzz**
Fixes #20070 (https://github.com/owncloud/core/issues/20070#issuecomment-154014521) for real

@DeepDiver1975 @LukasReschke @rullzer @MorrisJobke

### CODE REVIEW

**Comment by blizzz** [comment]
and @PVince81

File: lib/private/subadmin.php [hunk 1]
```diff
@@ -161,10 +167,14 @@ public function getAllSubAdmins() {
 
 		$subadmins = [];
 		while($row = $result->fetch()) {
-			$subadmins[] = [
-				'user'  => $this->userManager->get($row['uid']),
-				'group' => $this->groupManager->get($row['gid'])
-			];
+			$user = $this->userManager->get($row['uid']);
+			$group = $this->groupManager->get($row['gid']);
+			if(!is_null($user) && !is_null($group)) {
```
Comment (PVince81) [inline lib/private/subadmin.php]: Can it happen that one is null and not the other ? I guess it's probably correct to request both to be non-null in this specific case.

**Comment by PVince81** [comment]
Code looks good :+1:

File: lib/private/subadmin.php (hunk 1 above)
Comment (blizzz) [inline lib/private/subadmin.php]: Yes, an existing local group may have a vanished LDAP user as subadmin, for example

Comment (MorrisJobke) [inline lib/private/subadmin.php]: Should we then directly delete this entry?

Comment (blizzz) [inline lib/private/subadmin.php]: No, you do not know why the user is not there. Might be just temporary. Cleaning this up should be done with the post delete hooks.

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by karlitschek** [comment]
nice :+1:

**Pull request merged**